cmd/admission/app/configure/configure.go

```diff
@@ -79,78 +94,146 @@ func (c *Config) CheckPortOrDie() error {
 	return nil
 }
 
-// PatchMutateWebhookConfig patches a CA bundle into the specified webhook config.
-func PatchMutateWebhookConfig(client admissionregistrationv1beta1client.MutatingWebhookConfigurationInterface,
-	webhookConfigName, webhookName string, caBundle []byte) error {
-	config, err := client.Get(webhookConfigName, metav1.GetOptions{})
-	if err != nil {
-		return err
+func useGeneratedNameIfRequired(configured, generated string) string {
+	if configured != "" {
+		return configured
 	}
-	prev, err := json.Marshal(config)
+	return generated
+}
+
+// RegiesterWebhooks register webhooks for admission service
+func RegiesterWebhooks(c *Config, clienset *kubernetes.Clientset, cabundle []byte) error {
+	// Get the admission service
+	admissionService, err := clienset.CoreV1().Services(
+		c.AdmissionServiceNamespace).Get(c.AdmissionServiceName, metav1.GetOptions{})
 	if err != nil {
+		glog.Fatalf("Failed to get admission service %v\n", err)
 		return err
 	}
-	found := false
-	for i, w := range config.Webhooks {
-		if w.Name == webhookName {
-			config.Webhooks[i].ClientConfig.CABundle = caBundle[:]
-			found = true
-			break
-		}
-	}
-	if !found {
-		return apierrors.NewInternalError(fmt.Errorf(
-			"webhook entry %q not found in config %q", webhookName, webhookConfigName))
-	}
-	curr, err := json.Marshal(config)
-	if err != nil {
-		return err
+	serviceRef := metav1.NewControllerRef(admissionService, metav1.SchemeGroupVersion.WithKind("Service"))
+	ignorePolicy := v1beta1.Ignore
+
+	//Prepare validate webhooks
+	path := "/jobs"
+	JobValidateHooks := v1beta1.ValidatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: useGeneratedNameIfRequired(c.ValidateWebhookConfigName,
+				fmt.Sprintf(ValidateConfigName, c.AdmissionServiceName)),
+		},
+		Webhooks: []v1beta1.Webhook{{
+			Name: useGeneratedNameIfRequired(c.ValidateWebhookName, ValidateHookName),
+			Rules: []v1beta1.RuleWithOperations{
+				{
+					Operations: []v1beta1.OperationType{v1beta1.Create, v1beta1.Update},
+					Rule: v1beta1.Rule{
+						APIGroups:   []string{"batch.volcano.sh"},
+						APIVersions: []string{"v1alpha1"},
+						Resources:   []string{"jobs"},
+					},
+				},
+			},
+			ClientConfig: v1beta1.WebhookClientConfig{
+				Service: &v1beta1.ServiceReference{
+					Name:      c.AdmissionServiceName,
+					Namespace: c.AdmissionServiceNamespace,
+					Path:      &path,
+				},
+				CABundle: cabundle,
+			},
+			FailurePolicy: &ignorePolicy,
+		}},
 	}
-	patch, err := strategicpatch.CreateTwoWayMergePatch(prev, curr, v1beta1.MutatingWebhookConfiguration{})
-	if err != nil {
+	JobValidateHooks.OwnerReferences = append(JobValidateHooks.OwnerReferences, *serviceRef)
```

Comment:
No need setting this.

Reply:
for the purpose of deletion within admission service.

Reply:
When service is deleted, the configuration is also deleted, But when service is created, the configuration is not created auto.